File: src/velvet/automation/JobQueue.java
package velvet.automation;

import java.util.LinkedList;

public class JobQueue {

    private LinkedList<Runnable> jobs;

    public JobQueue(){
        jobs = new LinkedList<>();
    }

    public synchronized Runnable popJob(){
        if(jobs.isEmpty()){ return null; }
        return jobs.pop();
    }

    public synchronized void addJob(Runnable job){
        jobs.add(job);
    }
}

File: src/velvet/automation/JobQueue.kt
package velvet.automation

import java.util.*

class JobQueue {

    private val jobs: LinkedList<() -> Unit> = LinkedList()

    @Synchronized
    fun popJob(): (() -> Unit)? {
        if (jobs.isEmpty()) { return null; }
        return jobs.pop()
    }

    @Synchronized
    fun addJob(job: () -> Unit) = jobs.add(job)
}
